plugins/hls-alternate-number-format-plugin/test/Main.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeOperators  #-}
+{-# LANGUAGE ViewPatterns   #-}
+module Main ( main ) where
+
+import           Data.List                        (find)
+import           Data.Text                        (Text)
+import qualified Data.Text                        as T
+import qualified Ide.Plugin.AlternateNumberFormat as AlternateNumberFormat
+import qualified Ide.Plugin.Conversion            as Conversion
+import           System.FilePath                  ((</>))
+import           Test.Hls
+import           Text.Regex.TDFA                  ((=~))
+
+main :: IO ()
+main = defaultTestRunner test
+
+alternateNumberFormatPlugin :: PluginDescriptor IdeState
+alternateNumberFormatPlugin = AlternateNumberFormat.descriptor "alternateNumberFormat"
+
+test :: TestTree
+test = testGroup "alternateNumberFormat" [
+    codeActionHex "TIntDtoH" 3 13]
+
+testDataDir :: FilePath
+testDataDir = "test" </> "testdata"
+
+-- most helpers derived from explicit-imports-plugin Main Test file
+
+goldenAlternateFormat :: FilePath -> (TextDocumentIdentifier -> Session ()) -> TestTree
+goldenAlternateFormat fp = goldenWithHaskellDoc alternateNumberFormatPlugin (fp <> " (golden)") testDataDir fp "expected" "hs"
+
+codeActionTest :: (Maybe Text -> Bool) -> FilePath -> Int -> Int -> TestTree
+codeActionTest filter' fp line col = goldenAlternateFormat fp $ \doc -> do
+  actions <- getAllCodeActions doc
+  -- can't generate code actions?
+  liftIO $ assertBool "Actions is Not Null" (not $ null actions)
+--   case find (filter' . caTitle) actions of
```

Comment:
Could we uncomment here the core of the test to see what has to say the ci testing workflow?